xds/server_options.go

```
@@ -71,6 +73,27 @@ type ServingModeChangeArgs struct {
 //
 // Notice: This API is EXPERIMENTAL and may be changed or removed in a
 // later release.
-func BootstrapContentsForTesting(contents []byte) grpc.ServerOption {
-	return &serverOption{apply: func(o *serverOptions) { o.bootstrapContentsForTesting = contents }}
+func BootstrapContentsForTesting(bootstrapContents []byte) grpc.ServerOption {
+	config, err := bootstrap.NewConfigForTesting(bootstrapContents)
+	if err != nil {
+		return &serverOption{apply: func(o *serverOptions) { o.clientPoolForTesting = nil }}
```

Maybe throw an error log here so that the user knows why things are failing.

---

Added warning log because error log fails the test